doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,418 @@ Compliant example::
+expressions in for-loop headers.
+
+Compliant example::
+
+    uint32_t i;
+    
+    for (i = 0U; i < 5U; i++) {
+        printf("count: %d \n", i);
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t i;
+       
+       for (i = 0U;i < 5U;i++) {
+           printf("count: %d \n", i);
+       }
+
+
+
+Braces
+======
+
+BC-01: Braces after if/switch/for/do/while shall be in the same line
+--------------------------------------------------------------------
+
+The statement after if/switch/for/do/while shall always be a compound statement
+with its opening brace in the same line as the keyword.
+
+Compliant example::
+
+    uint32_t numerator = 32U;
+    uint32_t denominator = 0U;
+    uint32_t quotient;
+    
+    if (denominator != 0U) {
+        quotient = numerator / denominator;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t numerator = 32U;
+       uint32_t denominator = 0U;
+       uint32_t quotient;
+       
+       if (denominator != 0U)
+       {
+           quotient = numerator / denominator;
+       }
+
+
+BC-02: Newline and brace shall be after function prototypes
+-----------------------------------------------------------
+
+A function body shall start with a line containing a single opening brace.
+
+Compliant example::
+
+    uint32_t func_showcase(uint32_t param)
+    {
+        return param;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t func_showcase(uint32_t param) {
+           return param;
+       }
+
+
+
+Alignment
+=========
+
+AL-01: A `switch` statement and its subordinate `case` shall be aligned
+-----------------------------------------------------------------------
+
+Compliant example::
+
+    switch(suffix) {
+    case 'u':
+        do_something();
+        break;
+    default:
+        do_something_else();
+        break;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       switch(suffix) {
+           case 'u':
+               do_something();
+               break;
+           default:
+               do_something_else();
+               break;
+       }
+
+
+AL-02: Function parameters shall be aligned
+-------------------------------------------
+
+When function call parameters are not in single line, the parameters shall be
+aligned to the position which is lined up two space from function call.
+
+Compliant example::
+
+    uint32_t showcase;
+    
+    showcase = func(param_1,
+                 param_2,
+                 param_3);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase;
+       
+       showcase = func(param_1,
+                  param_2,
+                  param_3);
+
+
+
+Comments
+========
+
+CM-01: `//` shall not be used for single-line comments
+------------------------------------------------------
+
+`/*  */` shall be used to replace `//` for single-line comments.
```

Comment:
This can happen when using asterisks.  ReST treats ``*italics text*`` to mean *italics text*.  Try escaping the asterisk like this:
```
'/\*  \*/' shall be used to replace '//' for single-line comments.
```

Reply:
Thanks for the suggestion! Will update it in v4.